File: MVVM_ROOM/app/src/main/java/com/tyagiabhinav/loremipsum/di/RetrofitModule.java
package com.tyagiabhinav.loremipsum.di;

import com.tyagiabhinav.loremipsum.model.service.PostsAPI;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import dagger.hilt.InstallIn;
import dagger.hilt.components.SingletonComponent;
import retrofit2.Retrofit;
import retrofit2.converter.scalars.ScalarsConverterFactory;

@Module
@InstallIn(SingletonComponent.class)
public class RetrofitModule {

    @Singleton
    @Provides
    Retrofit providesRetrofit(){
        return new Retrofit.Builder()
                .baseUrl("https://jsonplaceholder.typicode.com")
                .addConverterFactory(ScalarsConverterFactory.create())
                .build();
    }

    @Singleton
    @Provides
    PostsAPI providesPostsAPI(Retrofit retrofit) {
        return retrofit.create(PostsAPI.class);
    }
}

File: MVVM_ROOM/app/src/main/java/com/tyagiabhinav/loremipsum/di/RetrofitModule.kt
package com.tyagiabhinav.loremipsum.di

import com.tyagiabhinav.loremipsum.model.service.PostsApi
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory
import javax.inject.Singleton

@Module
@InstallIn(SingletonComponent::class)
class RetrofitModule {

    @Singleton
    @Provides
    fun providesRetrofit(): Retrofit =  Retrofit.Builder()
            .baseUrl("https://jsonplaceholder.typicode.com")
            .addConverterFactory(GsonConverterFactory.create())
            .build()

    @Singleton
    @Provides
    fun providesPostsAPI(retrofit: Retrofit): PostsApi = retrofit.create(PostsApi::class.java)
}